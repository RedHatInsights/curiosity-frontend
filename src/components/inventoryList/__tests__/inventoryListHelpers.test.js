import {
  inventoryListHelpers,
  applySortFilters,
  parseInventoryFilters,
  parseRowCellsListData
} from '../inventoryListHelpers';
import {
  RHSM_API_QUERY_SORT_DIRECTION_TYPES as SORT_DIRECTION_TYPES,
  RHSM_API_QUERY_TYPES
} from '../../../types/rhsmApiTypes';

describe('InventoryListHelpers', () => {
  it('should have specific functions', () => {
    expect(inventoryListHelpers).toMatchSnapshot('inventoryListHelpers');
  });

  it('parseRowCellsListData should parse and return formatted/filtered table cells.', () => {
    const filters = [];
    const cellData = {
      lorem: 'ipsum',
      dolor: 'sit'
    };

    expect(parseRowCellsListData({ filters, cellData })).toMatchSnapshot('basic cell data');

    filters.push({
      id: 'lorem'
    });

    expect(parseRowCellsListData({ filters, cellData })).toMatchSnapshot('filtered data');

    filters[0] = {
      id: 'lorem',
      cell: {
        title: 'object, cell, lorem',
        props: { textCenter: true }
      }
    };

    expect(parseRowCellsListData({ filters, cellData })).toMatchSnapshot('custom cell data');

    filters[0] = {
      id: 'lorem',
      header: {
        title: 'object, header, lorem',
        props: { textCenter: true }
      }
    };

    expect(parseRowCellsListData({ filters, cellData })).toMatchSnapshot('custom header data');

    filters[0] = {
      header: ({ lorem, dolor }) => `${lorem.title}/${dolor.title}`,
      cell: ({ lorem, dolor }) => `${lorem.value}/${dolor.value}`
    };

    expect(parseRowCellsListData({ filters, cellData })).toMatchSnapshot('custom callback data');
  });

  it('parseInventoryFilters should parse and return updated filters for table cells', () => {
    const filters = [
      {
        id: 'lorem',
        isSortable: false
      }
    ];

    expect(parseInventoryFilters({ filters, onSort: () => {} })).toMatchSnapshot('NOT sortable');

    filters[0].isSortable = true;
    expect(parseInventoryFilters({ filters, onSort: () => {} })).toMatchSnapshot('sortable');
  });

  it('applySortFilters should apply and return updated filters for table sorting', () => {
    const filter = {
      id: 'lorem',
      isSortable: false
    };

    expect(applySortFilters({ filter, onSort: undefined })).toMatchSnapshot('NOT sortable');

    filter.isSortable = true;
    expect(applySortFilters({ filter, onSort: () => {} })).toMatchSnapshot('sortable');

    expect({
      ascending: applySortFilters({
        filter,
        onSort: () => {},
        query: { [RHSM_API_QUERY_TYPES.DIRECTION]: SORT_DIRECTION_TYPES.ASCENDING }
      }),
      descending: applySortFilters({
        filter,
        onSort: () => {},
        query: { [RHSM_API_QUERY_TYPES.DIRECTION]: SORT_DIRECTION_TYPES.DESCENDING }
      })
    }).toMatchSnapshot('sortable, direction');

    expect(
      applySortFilters({
        filter,
        onSort: () => {},
        query: {
          [RHSM_API_QUERY_TYPES.DIRECTION]: SORT_DIRECTION_TYPES.ASCENDING,
          [RHSM_API_QUERY_TYPES.SORT]: 'lorem'
        }
      })
    ).toMatchSnapshot('sortable, active column');
  });
});
