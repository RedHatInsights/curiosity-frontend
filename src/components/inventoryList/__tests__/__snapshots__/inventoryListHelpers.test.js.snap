// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InventoryListHelpers applySortFilters should apply and return updated filters for table sorting: NOT sortable 1`] = `
Object {
  "id": "lorem",
  "isSortable": false,
}
`;

exports[`InventoryListHelpers applySortFilters should apply and return updated filters for table sorting: sortable 1`] = `
Object {
  "id": "lorem",
  "isSortable": true,
  "onSort": [Function],
}
`;

exports[`InventoryListHelpers applySortFilters should apply and return updated filters for table sorting: sortable, active column 1`] = `
Object {
  "id": "lorem",
  "isSortable": true,
  "onSort": [Function],
  "sortActive": true,
  "sortDirection": "asc",
}
`;

exports[`InventoryListHelpers applySortFilters should apply and return updated filters for table sorting: sortable, direction 1`] = `
Object {
  "ascending": Object {
    "id": "lorem",
    "isSortable": true,
    "onSort": [Function],
    "sortDirection": "asc",
  },
  "descending": Object {
    "id": "lorem",
    "isSortable": true,
    "onSort": [Function],
    "sortDirection": "desc",
  },
}
`;

exports[`InventoryListHelpers applySortFilters should apply and return updated filters for table sorting: sortable, set a default field 1`] = `
Object {
  "id": "lorem",
  "isSortDefault": true,
  "isSortable": true,
  "onSort": [Function],
  "sortActive": true,
  "sortDefaultInitialDirection": "lorem",
  "sortDirection": "lorem",
}
`;

exports[`InventoryListHelpers applyWrappableFilters should apply and return updated filters for table header wrapping: NOT wrappable header 1`] = `
Array [
  Object {
    "id": "lorem",
    "isWrappable": false,
  },
]
`;

exports[`InventoryListHelpers applyWrappableFilters should apply and return updated filters for table header wrapping: append/push a wrappable transform 1`] = `
Object {
  "id": "lorem",
  "isWrappable": true,
  "transforms": Array [
    "lorem",
    [Function],
  ],
}
`;

exports[`InventoryListHelpers applyWrappableFilters should apply and return updated filters for table header wrapping: apply default wrappable transform 1`] = `
Object {
  "id": "lorem",
  "isWrappable": true,
  "transforms": Array [
    [Function],
  ],
}
`;

exports[`InventoryListHelpers applyWrappableFilters should apply and return updated filters for table header wrapping: wrappable header 1`] = `
Array [
  Object {
    "id": "lorem",
    "isWrappable": true,
    "transforms": Array [
      [Function],
    ],
  },
]
`;

exports[`InventoryListHelpers parseInventoryFilters should parse and return updated filters for table cells: NOT sortable 1`] = `
Array [
  Object {
    "id": "lorem",
    "isSortable": false,
  },
]
`;

exports[`InventoryListHelpers parseInventoryFilters should parse and return updated filters for table cells: sortable 1`] = `
Array [
  Object {
    "id": "lorem",
    "isSortable": true,
    "onSort": [Function],
  },
]
`;

exports[`InventoryListHelpers parseRowCellsListData should parse and return formatted/filtered table cells.: basic cell data 1`] = `
Object {
  "cells": Array [
    "ipsum",
    "sit",
  ],
  "columnHeaders": Array [
    "t(curiosity-inventory.header, {\\"context\\":\\"lorem\\"})",
    "t(curiosity-inventory.header, {\\"context\\":\\"dolor\\"})",
  ],
  "data": Object {
    "dolor": Object {
      "title": "t(curiosity-inventory.header, {\\"context\\":\\"dolor\\"})",
      "value": "sit",
    },
    "lorem": Object {
      "title": "t(curiosity-inventory.header, {\\"context\\":\\"lorem\\"})",
      "value": "ipsum",
    },
  },
}
`;

exports[`InventoryListHelpers parseRowCellsListData should parse and return formatted/filtered table cells.: custom callback data 1`] = `
Object {
  "cells": Array [
    Object {
      "title": "ipsum/sit",
    },
  ],
  "columnHeaders": Array [
    Object {
      "title": "t(curiosity-inventory.header, {\\"context\\":\\"lorem\\"})/t(curiosity-inventory.header, {\\"context\\":\\"dolor\\"})",
      "transforms": Array [],
    },
  ],
  "data": Object {
    "dolor": Object {
      "title": "t(curiosity-inventory.header, {\\"context\\":\\"dolor\\"})",
      "value": "sit",
    },
    "lorem": Object {
      "title": "t(curiosity-inventory.header, {\\"context\\":\\"lorem\\"})",
      "value": "ipsum",
    },
  },
}
`;

exports[`InventoryListHelpers parseRowCellsListData should parse and return formatted/filtered table cells.: custom cell data 1`] = `
Object {
  "cells": Array [
    Object {
      "props": Object {
        "textCenter": true,
      },
      "title": "object, cell, lorem",
    },
  ],
  "columnHeaders": Array [
    Object {
      "title": "t(curiosity-inventory.header, {\\"context\\":\\"lorem\\"})",
      "transforms": Array [],
    },
  ],
  "data": Object {
    "dolor": Object {
      "title": "t(curiosity-inventory.header, {\\"context\\":\\"dolor\\"})",
      "value": "sit",
    },
    "lorem": Object {
      "title": "t(curiosity-inventory.header, {\\"context\\":\\"lorem\\"})",
      "value": "ipsum",
    },
  },
}
`;

exports[`InventoryListHelpers parseRowCellsListData should parse and return formatted/filtered table cells.: custom header data 1`] = `
Object {
  "cells": Array [
    Object {
      "title": "ipsum",
    },
  ],
  "columnHeaders": Array [
    Object {
      "props": Object {
        "textCenter": true,
      },
      "title": "object, header, lorem",
      "transforms": Array [],
    },
  ],
  "data": Object {
    "dolor": Object {
      "title": "t(curiosity-inventory.header, {\\"context\\":\\"dolor\\"})",
      "value": "sit",
    },
    "lorem": Object {
      "title": "t(curiosity-inventory.header, {\\"context\\":\\"lorem\\"})",
      "value": "ipsum",
    },
  },
}
`;

exports[`InventoryListHelpers parseRowCellsListData should parse and return formatted/filtered table cells.: filtered data 1`] = `
Object {
  "cells": Array [
    Object {
      "title": "ipsum",
    },
  ],
  "columnHeaders": Array [
    Object {
      "title": "t(curiosity-inventory.header, {\\"context\\":\\"lorem\\"})",
      "transforms": Array [],
    },
  ],
  "data": Object {
    "dolor": Object {
      "title": "t(curiosity-inventory.header, {\\"context\\":\\"dolor\\"})",
      "value": "sit",
    },
    "lorem": Object {
      "title": "t(curiosity-inventory.header, {\\"context\\":\\"lorem\\"})",
      "value": "ipsum",
    },
  },
}
`;

exports[`InventoryListHelpers should have specific functions: inventoryListHelpers 1`] = `
Object {
  "applySortFilters": [Function],
  "applyWrappableFilters": [Function],
  "parseInventoryFilters": [Function],
  "parseRowCellsListData": [Function],
}
`;
